internal/tools/changes/modules.go

```diff
@@ -34,5 +46,158 @@ func getModFile(path string) (*modfile.File, error) {
 }
 
 func shortenModPath(modulePath string) string {
-	return strings.TrimPrefix(modulePath, sdkRepoPath)
+	if modulePath == sdkRepo {
+		return rootModule
+	}
+
+	modulePath = strings.TrimLeft(modulePath, "/")
+	return strings.TrimPrefix(modulePath, sdkRepo+"/")
+}
+
+func lengthenModPath(modulePath string) string {
+	if modulePath == rootModule {
+		return sdkRepo
+	}
+
+	return sdkRepo + "/" + modulePath
+}
+
+// discoverModules returns a list of all modules and a map between all packages and their providing module.
+func discoverModules(golist golist.ModuleClient, root string) ([]string, map[string]string, error) {
+	var modules []string
+	packages := map[string]string{}
+
+	err := filepath.Walk(root, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if info.Name() == "go.mod" {
+			modFile, err := getModFile(path)
+			if err != nil {
+				return err
+			}
+
+			mod := shortenModPath(modFile.Module.Mod.String())
+
+			modPackages, err := golist.Packages(mod)
+			if err != nil {
+				return err
+			}
+
+			for _, p := range modPackages {
+				packages[p] = mod
+			}
+
+			modules = append(modules, mod)
+		} else if info.Name() == "testdata" && path != "testdata" {
+			// skip testdata directory unless it is the root directory.
+			return filepath.SkipDir
+		}
+
+		return nil
+	})
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return modules, packages, nil
+}
+
+func updateDependencies(repoPath, mod string, dependencies []string, enc *VersionEnclosure) error {
+	goModPath := filepath.Join(repoPath, mod, "go.mod")
+
+	modFile, err := getModFile(goModPath)
+	if err != nil {
+		return fmt.Errorf("couldn't update %s's dependencies: %v", mod, err)
+	}
+
+	for _, dep := range dependencies {
+		err = modFile.AddRequire(lengthenModPath(dep), enc.ModuleVersions[dep].Version)
+		if err != nil {
+			return fmt.Errorf("couldn't update %s's dependency on %s: %v", mod, dep, err)
+		}
+	}
+
+	out, err := modFile.Format()
+	if err != nil {
+		return fmt.Errorf("couldn't update %s's dependencies: %v", mod, err)
+	}
+
+	return util.WriteFile(out, goModPath, false)
+}
+
+// defaultVersion returns a default version for the given module based on its import path. If a version suffix /vX is
+// present in the import path, the default version will be vX.0.0. Otherwise, the version will be v0.0.0.
```

Comment:
It looks like there is precedent for starting at `v0.1.0` (`go-cmp` did this, for example). I think you're right that it could be confusing, so the tool now defaults to `v0.1.0`, but still `vX.0.0` when X > 0.